ruff_lsp/server.py

```diff
@@ -1002,7 +1005,22 @@ def _get_settings_by_document(document: workspace.Document | None) -> WorkspaceS
 ###
 
 
-def _executable_path(settings: WorkspaceSettings) -> str:
+def _find_ruff_binary(settings: WorkspaceSettings) -> str:
+    """Returns the path to the executable, checking that is has the required minimum
+    version."""
+    executable = _find_ruff_binary_path(settings)
+
+    # This call is cached so it should generally be free
+    version = _executable_version(executable)
+    if not VERSION_REQUIREMENT.contains(version, prereleases=True):
+        message = f"ruff {VERSION_REQUIREMENT} required, but {executable} is {version}"
+        log_error(message)
+        raise RuntimeError(message)
```

Comment:
I think this should be enforced on a per-subcommand basis -- e.g., we should continue to allow older Ruff versions for all the lint commands, but should error if they try to use the formatter.